cmd/cmd-exec/pkg/commands/commands.go

```diff
@@ -105,8 +115,15 @@ func runCmd(command Command, ei *server.ExecutionInfo) error {
 	cmd.Stdout = ei.Log
 	cmd.Stderr = ei.Log
 
-	curEnvs := append(os.Environ(), fmt.Sprintf("HOME=%s", ei.TmpDir))
-	cmd.Env = curEnvs
+	envs := make([]string, 0)
+	envs = append(envs, fmt.Sprintf("HOME=%s", ei.TmpDir))
+	envs = append(envs, os.Environ()...)
+
+	for i := range command.Envs {
+		envs = append(envs, command.Envs[i].toKV())
+	}
+
+	cmd.Env = envs
```

Comment:
Minor question to think about: is this the best order to append variables? There's no attempt at deduplication here, which is fine. Environment variables can be overwritten by what comes last in the order. Do we want users to be able to override HOME and os.Environ()?

Reply:
os.Environ contains the "normal" envs which is probably not great but it contains the one from the function definition as well and filtering out based on assumed naming like "if they start with KUBERNETES_" delete them is not a great idea. The additional ones are not a security issue so that is the best way to do it.